rewrite-core/src/main/java/org/openrewrite/RecipeScheduler.java

```diff
@@ -111,11 +123,29 @@ default RecipeRun scheduleRun(Recipe recipe,
             return recipeRun.withDataTables(ctx.getMessage(ExecutionContext.DATA_TABLES, emptyMap()));
         }
 
+        List<Result> results = createAndProcessResults(
+                before,
+                after,
+                ctx,
+                recipeThatAddedOrDeletedSourceFile
+        );
+
+        runStatsTable.record(ctx, recipe, runStats);
+        return recipeRun
+                .withResults(results)
+                .withDataTables(ctx.getMessage(ExecutionContext.DATA_TABLES, emptyMap()));
+    }
+
+    static List<Result> createAndProcessResults(
+            List<? extends SourceFile> before,
+            List<? extends SourceFile> after,
+            ExecutionContext ctx,
+            Map<UUID, Stack<Recipe>> recipeThatAddedOrDeletedSourceFile
+    ) {
```

Comment:
This should theoretically be added to the `RecipeSchedulerUtils` but I wanted to simplify the review of this diff